BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -147,6 +147,184 @@ public static Vector3 GetAirToGroundTarget(Vector3 targetPosition, Vector3 targe
+                        Vector3.ProjectOnPlane(targetPosition - missileVessel.transform.position, upDirection).normalized;
+
+                    // Altitude clamp based on rangeFactor and maxAlt, cannot be lower than target
+                    float altitudeClamp = Mathf.Clamp(targetAlt + rangeFactor * Vector3.Dot(targetPosition - missileVessel.transform.position,planarDirectionToTarget), targetAlt, maxAltitude);
+
+                    // Old loft climb logic, wanted to limit turn. Didn't work well but leaving it in if I decide to fix it
+                    /*if (missileVessel.altitude < (altitudeClamp - 0.5f))
+                    //gain altitude if launching from stationary
+                    {*/
+                        //currSpeed = (float)missileVessel.Velocity().magnitude;
+
+                        // 5g turn, v^2/r = a, v^2/(dh*(tan(45°/2)sin(45°))) > 5g, v^2/(tan(45°/2)sin(45°)) > 5g * dh, I.E. start turning when you need to pull a 5g turn,
+                        // before that the required gs is lower, inversely proportional
+                        /*if (loftState == 1 || (currSpeed * currSpeed * 0.2928932188134524755991556378951509607151640623115259634116f) >= (5f * (float)PhysicsGlobals.GravitationalAcceleration) * (altitudeClamp - missileVessel.altitude))
+                        {*/
+                            /*
+                            loftState = 1;
+
+                            // Calculate upwards and forwards velocity components
+                            velUp = Vector3.Dot(missileVessel.Velocity(), upDirection);
+                            velForwards = (float)(missileVessel.Velocity() - upDirection * velUp).magnitude;
+
+                            // Derivation of relationship between dh and turn radius
+                            // tan(theta/2) = dh/L, sin(theta) = L/r
+                            // tan(theta/2) = sin(theta)/(1+cos(theta))
+                            float turnR = (float)(altitudeClamp - missileVessel.altitude) * (currSpeed * currSpeed + currSpeed * velForwards) / (velUp * velUp);
+
+                            float accel = Mathf.Clamp(currSpeed * currSpeed / turnR, 0, 5f * (float)PhysicsGlobals.GravitationalAcceleration);
+                            */
+
+                            // Limit climb angle by turnFactor, turnFactor goes negative when above target alt
+                            float turnFactor = (float)(altitudeClamp - missileVessel.altitude) / (4f * (float)missileVessel.srfSpeed);
+                            turnFactor = Mathf.Clamp(turnFactor, -1f, 1f);
+                            if (BDArmorySettings.DEBUG_MISSILES) Debug.Log("[BDArmory.MissileLauncher]: AAM Loft altitudeClamp: [" + altitudeClamp + "] COS: [" + Mathf.Cos(loftAngle * turnFactor * Mathf.Deg2Rad) + "], SIN: [" + Mathf.Sin(loftAngle * turnFactor * Mathf.Deg2Rad) + "], turnFactor: [" + turnFactor + "].");
```

Comment:
Instead of adding strings (which requires multiple memory allocations), prefer string interpolation, i.e.,
```csharp
if (BDArmorySettings.DEBUG_MISSILES) Debug.Log($"[BDArmory.MissileGuidance]: AAM Loft altitudeClamp: [{altitudeClamp:G6}] COS: [{Mathf.Cos(loftAngle * turnFactor * Mathf.Deg2Rad):G3}], SIN: [{Mathf.Sin(loftAngle * turnFactor * Mathf.Deg2Rad):G3}], turnFactor: [{turnFactor:G3}].");
```

Reply:
Done for all debug statements.